types/loopback/index.d.ts

```diff
@@ -1262,15 +1345,34 @@ declare namespace l {
              * @param {any} info Additional information about the command outcome.
              * @param {number} info.count number of instances (rows, documents) destroyed
              */
-            static destroyAll(where?: any, callback?: (err: Error, info: any, infoCount: number) => void): Promise<{ info: any, infoCount: number}> | void;
+            static destroyAll(where: any, callback: CallbackWithMultipleResults<any, number>): void;
+
+            /**
+             * Destroy all model instances that match the optional `where` specification
+             * @param {any} [where] Optional where filter, like:
+             * ```
+             * {key: val, key2: {gt: 'val2'}, ...}
+             * ```
+             * <br/>See
+             * [Where filter](docs.strongloop.com/display/LB/Where+filter#Wherefilter-Whereclauseforothermethods)
+             *
+             * @returns {Promise<{ count: number }>} number of instances (rows, documents) destroyed
+             */
+            static destroyAll(where?: any): Promise<{ info: any, infoCount: number}>;
```

Comment:
I know it's annoying, but would you be able to move the callback versions below the `Promise` versions of these overloads? That way things like signature help will actually work out better.

Reply:
What about for functions with 2+ overloads? Is the rule that one should go from most specific to least specific?

Reply:
Well, it sort of depends. For arity, I think the idea is to go with fewer first. but you know, this is silly, this can be adjusted in a later iteration.